src/server.cpp

```diff
@@ -2,73 +2,71 @@
+        }
+        return "uncnown";
     }
-    return "uncnown";
-}
-
-Server::Server(boost::asio::io_service& io_service, const std::string& secret)
-    : m_socket(io_service, udp::endpoint(udp::v4(), 9999)),
-      m_secret(secret)
-{
-}
-
-void Server::asyncReceive(const std::function<void(const error_code&, const std::optional<Packet>&)>& callback)
-{
-    m_socket.async_receive_from(boost::asio::buffer(m_recvBuffer), m_remoteEndpoint,
-       [this, callback](const error_code& error, std::size_t bytes) {handleReceive(error, bytes, callback);});
-}
-
-void Server::asyncSend(const Packet& response, const std::function<void(const error_code&)>& callback)
-{
-    m_socket.async_send_to(boost::asio::buffer(response.makeSendBuffer(m_secret)), m_remoteEndpoint,
-       [this, callback](const error_code& ec, std::size_t /*bytesTransferred*/) {handleSend(ec, callback);});
-}
 
-void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const std::optional<Packet>&)> callback)
-{
-        std::optional<Packet> oEmpty;
-        std::optional<Packet> oRequest;
+    Server::Server(boost::asio::io_service& io_service, const std::string& secret)
+        : m_socket(io_service, udp::endpoint(udp::v4(), 9999)),
+          m_secret(secret)
+    {
+    }
 
-        if (bytes < 20)
-            callback(lib::Error::numberBytesLess20, oEmpty);
+    void Server::asyncReceive(const std::function<void(const error_code&, const std::optional<Packet>&)>& callback)
+    {
+        m_socket.async_receive_from(boost::asio::buffer(m_recvBuffer), m_remoteEndpoint,
+           [this, callback](const error_code& error, std::size_t bytes) {handleReceive(error, bytes, callback);});
+    }
 
-    try
+    void Server::asyncSend(const Packet& response, const std::function<void(const error_code&)>& callback)
     {
-        const Packet request = Packet(m_recvBuffer, bytes, m_secret);
-        oRequest = request;
+        m_socket.async_send_to(boost::asio::buffer(response.makeSendBuffer(m_secret)), m_remoteEndpoint,
+           [this, callback](const error_code& ec, std::size_t /*bytesTransferred*/) {handleSend(ec, callback);});
     }
-    catch (const std::runtime_error& exception)
+
+    void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const std::optional<Packet>&)> callback)
     {
-        if (std::string (exception.what()) == "lib::Error::numberBytesLess20")
-            callback(lib::Error::numberBytesLess20, oEmpty);
+        if (bytes < 20)
+            callback(Error::numberOfBytesIsLessThan20, std::nullopt);
+        try
+        {
+            const Packet request = Packet(m_recvBuffer, bytes, m_secret);
+            callback(error, std::make_optional<Packet>(request));
+        }
+        catch (const Exception& exception)
+        {
+            if (exception.getErrorCode() == Error::numberOfBytesIsLessThan20)
```

Comment:
Why do you perform dispatch on error code? You can pass `exception.getErrorCode()` directly into the callback.